hovercraft/generate.py

```diff
@@ -107,13 +109,16 @@ def generate(args):
         template_info.add_resource(args.css, CSS_RESOURCE, target=target_path, extra_info='all')
         source_files.append(args.css)
 
-    # Make the resulting HTML
-    htmldata = rst2html(args.presentation, template_info, args.auto_console, args.skip_help,
-                        args.skip_notes)
+    # Make the resulting HTML (generation is relative to the basedir)
+    saved_cwd = os.getcwd()
+    os.chdir(args.basedir)
```

Comment:
I agree with you this isn't the cleanest solution.

However, how can the server process change the current directory while regenerating? I don't think you get any errors when using the web-server.
If in the shell, after starting the web-server, you put the job in background and manually change the directory, the job isn't still affected (ofc).

I think the cleanest solution would be staying always in the base directory, because if we don't want to change the target directly frequently and we want to stay just in one place, we **need** to stay in the root directory of the presentation, since it's the directory being considered by rst when resolving relative paths.